cardinal/ecs/engine.go

```diff
@@ -92,6 +93,78 @@ type Engine struct {
 	shutdownMutex sync.Mutex
 }
 
+func (e *Engine) ConsumeEVMMsgResult(evmTxHash string) ([]byte, []error, string, bool) {
+	rcpt, exists := e.consumeEVMMsgResult(evmTxHash)
+	return rcpt.ABIResult, rcpt.Errs, rcpt.EVMTxHash, exists
+}
+
+func (e *Engine) HandleEVMQuery(name string, abiRequest []byte) ([]byte, error) {
```

Comment:
i generally dont want to differentiate EVM/non-EVM once it has made its way to engine. the responsibility of managing the EVM specifics should be router's responsibility.

Reply:
It's a completely different data format though. Requests come through as ABI encoded bytes. Router has no idea what to do with those bytes unless it has it's query object, which would require importing ECS.

Reply:
I moved out the Query interface out of cardinal/ecs in my PR, can do it later, but the same way you are doing `GetMessageByName` there should be nothing stopping us from just doing that for query too.

Reply:
yup, i would have done the same pattern had i not needed to import ECS to deal with queries. ill open a ticket to do that once your PR is in.